index.js

```diff
@@ -58,12 +57,19 @@ function ShareDbMongo(mongo, options) {
   if (typeof mongo === 'string' || typeof mongo === 'function') {
     // We can only get the mongodb client instance in a callback, so
     // buffer up any requests received in the meantime
-    this.mongo = null;
-    this._mongoClient = null;
-    this.mongoPoll = null;
-    this._mongoPollClient = null;
-    this.pendingConnect = [];
-    this._connect(mongo, options);
+    var connection = this._connect(mongo, options);
+    this.mongo = connection.then(function(result) {
```

Comment:
This seems to be changing from a `DB` instance to a `Promise<DB>`. Since this isn't an underscore-prefixed property, I'd like to avoid this being a breaking change if possible. The published driver defers populating `mongo` anyways. The promise versions can go onto new (internal?) variables if needed.

Same for `mongoPoll`.

The underscore-prefixed properties are fine to change to promises, though we should consider renaming them.

Reply:
On that note, `this.pendingConnect` wasn't underscored; does that make this breaking anyway if I remove that...?